.circleci/config.yml

```diff
@@ -3,7 +3,9 @@ version: 2.1
 executors:
   gradle_docker:
     docker:
-      - image: cimg/openjdk:11.0
+      # This image isn't used to build Java project, rather to invoke Docker with Gradle.
+      # Using latest JDK image will give more recent Docker, albeit slightly behind cimg/base
```

Comment:
Since we're using a remote docker anyway, is this change related to upgrading docker? Does the client run locally by the build need to match the remote?

Reply:
I'm going to be swimming in pull requests, and it is odd to use old JDKs especially to run gradle docker commands. If possible, I'd like to update this, instead of another few dozen PRs later to do the same.

Reply:
changed the comment, but I'd like to leave the version update so that we have less work later. there are many many many versions behind due to a culture of not updating, which is a problem we need to address.

Reply:
My hesitation on 14 is mainly based on it going EOL in three weeks (I still think java's release/support model is ridiculous...). If we're going to be updating to 15 at that time, it doesn't seem super beneficial to update to 14 now, but maybe it'll flush out some issues so 👍

Reply:
Luckily, it doesn't have a built-in bladerunneresque expiry either! I don't think anyone will expect people to switch to 15 the day it is released. In fact, a lot of projects have a matrix of many JDKs. In practice, they linger a bit and I think that's acceptable.

Indeed, the highest value is flushing out issues so we don't have a massive pile of them later. If we find things we'll know as they occur when the JDK is released.